### PR TITLE
feat(semantic-release): semantic-release deployment to pypi and gh-pages

### PR DESCRIPTION
**Pull request by AntoineDao**

### CODE REVIEW

**Comment by AntoineDao** [comment]
Tried to use windows os above and received the following message:

```shell
The language 'python' is currently unsupported on the Windows Build Environment.
```

I'm speechless...

**Comment by AntoineDao** [comment]
If we're happy with this we should definitely squash and merge it to avoid showing my constant iterating to the world...

**Comment by AntoineDao** [comment]
Also if we are happy with this don't forget to add the following variables in travis:
* GH_TOKEN
* PYPI_PASSWORD
*PYPI_USERNAME

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,36 @@
+language: python
+
+python: 
+- "2.7"
+- "3.6"
```
Comment (mostaphaRoudsari) [inline .travis.yml]: not sure if it will work with Python 3 but since there is no tests it should pass! :|

File: deploy.sh [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+deploy_to_pypi() {
+  echo "Building distribution"
+  python setup.py sdist bdist_wheel
+  echo "Pushing new version to PyPi"
+  twine upload dist/* -u $PYPI_USERNAME -p $PYPI_PASSWORD 
+}
+
+build_docs() {
+  echo "Building documentation files"
+  sphinx-apidoc -f -e -d 4 -o ./docs ./honeybee
```
Comment (mostaphaRoudsari) [inline deploy.sh]:
```suggestion
  sphinx-apidoc -f -e -d 4 -o ./docs ./butterfly
```

File: setup.py [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+    author="Ladybug Tools",
+    author_email="info@ladybug.tools",
+    description="A light python API for creating and running OpenFoam cases for CFD simulation.",
+    long_description=long_description,
+    long_description_content_type="text/markdown",
+    url="https://github.com/ladybug-tools/butterfly",
+    packages=setuptools.find_packages(exclude=["tests"]),
+    classifiers=[
+        "Programming Language :: Python :: 2.7",
+        "Programming Language :: Python :: 3.6",
```
Comment (mostaphaRoudsari) [inline setup.py]: It is not really Python 3 compatible. Let's remove this for now until we write some tests and make it Python 3 compatible!

File: tox.ini [hunk 4]
```diff
@@ -0,0 +1,11 @@
+[tox]
+envlist = py27, py36
+
+[testenv]
+commands = pytest --cov=ladybug tests/
```
Comment (mostaphaRoudsari) [inline tox.ini]:
```suggestion
commands = pytest --cov=butterfly tests/
```

File: tox.ini [hunk 5]
```diff
@@ -0,0 +1,11 @@
...
+deps =
+    pytest
+    pytest-cov
+    euclid3
```
Comment (mostaphaRoudsari) [inline tox.ini]: Butterfly doesn't need `euclid`!

**Comment by mostaphaRoudsari** [comment]
> Also if we are happy with this don't forget to add the following variables in travis:
>
>    GH_TOKEN
>    PYPI_PASSWORD
>    PYPI_USERNAME

I don't know if you did it or I have done it before but this is already taken care of.

File: .travis.yml (hunk 1 above)
Comment (AntoineDao) [reply .travis.yml]: Will keep it for now :stuck_out_tongue:

File: deploy.sh (hunk 2 above)
Comment (AntoineDao) [reply deploy.sh]: Done.

File: setup.py (hunk 3 above)
Comment (AntoineDao) [reply setup.py]: Done.

File: tox.ini (hunk 5 above)
Comment (AntoineDao) [reply tox.ini]: Done.

**Pull request merged**